yarp/yarp_compiler.c

```diff
@@ -1320,7 +1323,7 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
           yp_statements_node_t *statements_node = (yp_statements_node_t *) node;
           yp_node_list_t node_list = statements_node->body;
           for (size_t index = 0; index < node_list.size - 1; index++) {
-              yp_compile_node(iseq, node_list.nodes[index], ret, src, !popped, compile_context);
+              yp_compile_node(iseq, node_list.nodes[index], ret, src, true, compile_context);
```

Comment:
This was wrong previously - we should pop all instructions on statements except the last one.